zavod/zavod/tests/tools/test_export_index.py

```diff
@@ -0,0 +1,37 @@
+from zavod.meta import get_catalog, load_dataset_from_path, Dataset
+from zavod.tools.meta_index import export_index
+from ..conftest import FIXTURES_PATH
+from zavod.context import Context
+from zavod import settings
+from json import load
+from zavod.runner import run_dataset
+from zavod.exporters import export
+
+DATASET_2_YML = FIXTURES_PATH / "test_dataset_2" / "test_dataset_2.yml"
+COLLECTION_YML = FIXTURES_PATH / "collection.yml"
+
+
+def test_export_index(vdataset: Dataset):
+    # Create dataset index files
+    context1 = Context(vdataset)
+    run_dataset(vdataset)
+    export(vdataset.name)
+
+    dataset2 = load_dataset_from_path(DATASET_2_YML)
+    run_dataset(dataset2)
+    export(dataset2.name)
+
+    # Clear catalog as if this is a fresh process separate from the earlier exports
+    get_catalog.cache_clear()
+
+    collection = load_dataset_from_path(COLLECTION_YML)
+    export(collection.name)
+    export_index(collection)
```

Comment:
Is this horrid or reasonable?

My thinking here is that global index is particularly interesting when it's run for a collection, and a collection with more than one dataset. For it to find the datasets, they need to at least have index files in their data dirs. And I want issues too, so I ended up just running them and full-blown exporting them.

Earlier iterations tried to be more precise about what the prerequisites are but now I just want something working.

Reply:
I think this is really good, we should have way more `get_catalog.cache_clear()` in places :)